### PR TITLE
Update kube-openapi to use validation

### PR DESCRIPTION
**Pull request by apelisse**
**What this PR does / why we need it**: Moves openapi validation code to kube-openapi, so that we can move the rest of the code to apimachinery repository, so that later we can use it from both the client and the server.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #Nothing

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"ImportPath": "k8s.io/kubernetes",
-	"GoVersion": "go1.9",
+	"GoVersion": "go1.8",
```
Comment (mengqiy) [inline Godeps/Godeps.json]: Why?

File: vendor/BUILD [hunk 2]
```diff
@@ -41,8 +41,6 @@ filegroup(
         "//vendor/github.com/aws/aws-sdk-go/service/elb:all-srcs",
         "//vendor/github.com/aws/aws-sdk-go/service/kms:all-srcs",
         "//vendor/github.com/aws/aws-sdk-go/service/sts:all-srcs",
-        "//vendor/github.com/aws/aws-sdk-go/vendor/github.com/go-ini/ini:all-srcs",
```
Comment (mengqiy) [inline vendor/BUILD]: Seems unrelated.

**Review by mengqiy — COMMENTED** [review]
Other LGTM

**Comment by mengqiy** [comment]
/assign @monopole 
for approval

File: Godeps/Godeps.json (hunk 1 above)
Comment (apelisse) [reply Godeps/Godeps.json]: Oh, OK. I need to update my go version.

File: vendor/BUILD (hunk 2 above)
Comment (apelisse) [reply vendor/BUILD]: I know but:
1. I don't control what `godep` does :-)
2. Less vendored code is better

File: pkg/kubectl/cmd/util/openapi/validation/validation.go [hunk 3]
```diff
@@ -24,20 +24,25 @@ import (
...
 
+// NewSchemaValidation creates a new SchemaValidation that can be used
+// to validate objects.
 func NewSchemaValidation(resources openapi.Resources) *SchemaValidation {
 	return &SchemaValidation{
 		resources: resources,
 	}
 }
 
+// ValidateBytes will validates the object against using the Resources
```
Comment (monopole) [inline pkg/kubectl/cmd/util/openapi/validation/validation.go]: s/will//

**Review by monopole — APPROVED** [review]
/lgtm
